src/app/components/restaurant-details/restaurant-details.component.ts

```diff
@@ -0,0 +1,72 @@
+import { Component, OnInit } from '@angular/core';
+import { map, Observable } from 'rxjs';
+import { ItemCardInterface } from '../../shared/interfaces/restaurant-card.interface';
+import { ActivatedRoute } from '@angular/router';
+import { CommonModule } from '@angular/common';
+import { MatIconModule } from '@angular/material/icon';
+import { ReviewFormatterPipe } from '../../shared/pipes/review-formatter.pipe';
+import { CardConfigInterface, PrimaryCardComponent } from '../primary-card/primary-card.component';
+
+@Component({
+  selector: 'fd-restaurant-details',
+  imports: [CommonModule, MatIconModule, ReviewFormatterPipe, PrimaryCardComponent],
+  templateUrl: './restaurant-details.component.html',
+  styleUrl: './restaurant-details.component.scss'
+})
+export class RestaurantDetailsComponent implements OnInit{
+  restaurant$!: Observable<ItemCardInterface>;
+
+  constructor(private route: ActivatedRoute) {}
+
+  ngOnInit(): void {
+    this.restaurant$ = this.route.data.pipe(map(data => data['restaurant']));
```

Comment:
```suggestion
    this.restaurant = this.route.snapshot.data.restaurant;
```
Instead of Observable use `restaurant` as synchronous code